src/main/java/javaslang/collection/Stream.java

```diff
@@ -1310,6 +1311,16 @@ public T head() {
         }
 
         @Override
+        public Tuple2<Stream<T>, Stream<T>> splitAt(int n) {
+            if(n <= 0) {
+                return Tuple.of(empty(), this);
+            } else {
+                Tuple2<Stream<T>, Stream<T>> t = tail().splitAt(n - 1);
+                return Tuple.of(new Cons<>(head, () -> t._1), t._2);
+            }
+        }
```

Comment:
For Stream the Tuple.of(take(n), drop(n)) solution is the best, because take() is lazy and no intermediate Tuple objects are required.